tests/newlinemanager_test.cpp

```diff
@@ -235,8 +235,8 @@ TEST_CASE("Get columns wigdet test", "[newlinemanager]")
 
     NewLineManager newLineMgr{ &canRawSender, true, nlmFactoryMock.get() };
 
-    for (NewLineManager::ColName ii : NewLineManager::ColNameIterator()) {
-        CHECK(newLineMgr.GetColsWidget(ii) != nullptr);
+    for (NewLineManager::ColName ii : NewLineManager::ColNameIterator{NewLineManager::ColName::IdLine}) {
+        CHECK(newLineMgr.GetColsWidget(NewLineManager::ColNameIterator{ii}) != nullptr);
```

Comment:
Can we have a operator-> implemented for Iterator to avoid helper functions? Isn't it required to be STL compliant?

Reply:
Arrow operator is out of iterator scope. Where do you want to use `->` here?